packages/manager/apps/catalog/src/components/Filters/Filters.tsx

```diff
@@ -31,12 +32,28 @@ const Filters: React.FC<FiltersProps> = ({
   const [selectedUniverses, setSelectedUniverses] = useState<string[]>([]);
   const [hasInteracted, setHasInteracted] = useState<boolean>(false);
 
+  const location = useLocation();
+
   const setFilters = () => {
     setParentSelectedCategories(selectedCategories);
     setParentSelectedUniverses(selectedUniverses);
     onApply();
   };
 
+  useEffect(() => {
+    const params = new URLSearchParams(location.search);
+    const universesURL = params.get('universes');
+    const categoriesURL = params.get('categories');
+    if (categoriesURL) {
+      setSelectedCategories(categoriesURL.split(','));
+      setParentSelectedCategories(categoriesURL.split(','));
+    }
+    if (universesURL) {
+      setSelectedUniverses(universesURL.split(','));
+      setParentSelectedUniverses(universesURL.split(','));
+    }
+  }, [location.search, products]);
```

Comment:
why put products in the dependency array ?

Reply:
it's an asynchronous effect, we get `location.search` before the 2API request that load the products, adding `products` seems to fix this issue. That's why.

Reply:
You didn't describe the issue actually
What is the issue ?

Reply:
If the variable are irrelevant while the products are not fetched, there should be a condition if(products.length > 0) for example

Reply:
I did many tests, products variable is mandatory in this useEffect as it is important that all the products are fetched to apply the filters from params from URL. I added your recommendation of `if(products.length > 0)` in filters.tsx and also in index.tsx is it important that all the products are fetched before we apply the params from the checkboxes and input.